doc/v5.0/man3/MPI_Pack_external_size.3
.\" -*- nroff -*-
.\" Copyright 2010 Cisco Systems, Inc.  All rights reserved.
.\" Copyright 2006-2008 Sun Microsystems, Inc.
.\" Copyright (c) 1996 Thinking Machines Corporation
.\" Copyright (c) 2020      Google, LLC. All rights reserved.
.\" $COPYRIGHT$
.TH MPI_Pack_external_size 3 "Sep 30, 2021" "5.0.0rc1" "Open MPI"

.SH NAME
\fBMPI_Pack_external_size\fP \- Calculates upper bound on space needed
to write to a portable format

.SH SYNTAX
.ft R

.SH C Syntax
.nf
#include <mpi.h>
int MPI_Pack_external_size(char *\fIdatarep\fP, int \fIincount\fP,
	MPI_Datatype \fIdatatype\fP, MPI_Aint *\fIsize\fP)

.fi
.SH Fortran Syntax
.nf
USE MPI
! or the older form: INCLUDE 'mpif.h'
MPI_PACK_EXTERNAL_SIZE(\fIDATAREP, INCOUNT, DATATYPE, SIZE, IERROR\fP)

	INTEGER		\fIINCOUNT, DATATYPE, IERROR\fP
	INTEGER(KIND=MPI_ADDRESS_KIND) \fISIZE\fP
	CHARACTER*(*)	\fIDATAREP\fP

.fi
.SH Fortran 2008 Syntax
.nf
USE mpi_f08
MPI_Pack_external_size(\fIdatarep\fP, \fIincount\fP, \fIdatatype\fP, \fIsize\fP, \fIierror\fP)
	TYPE(MPI_Datatype), INTENT(IN) :: \fIdatatype\fP
	INTEGER, INTENT(IN) :: \fIincount\fP
	CHARACTER(LEN=*), INTENT(IN) :: \fIdatarep\fP
	INTEGER(KIND=MPI_ADDRESS_KIND), INTENT(OUT) :: \fIsize\fP
	INTEGER, OPTIONAL, INTENT(OUT) :: \fIierror\fP

.fi
.SH INPUT PARAMETERS
.ft R
.TP 1i
datarep
Data representation (string).
.TP 1i
incount
Number of input data items (integer).
.TP 1i
datatype
Datatype of each input data item (handle).

.SH OUTPUT PARAMETERS
.ft R
.TP 1i
size
Upper bound on size of packed message, in bytes (integer).
.TP 1i
IERROR
Fortran only: Error status (integer).

.SH DESCRIPTION
.ft R
MPI_Pack_external_size allows the application to find out how much
space is needed to pack a message in the portable format defined by
the MPI Forum. It returns in \fIsize\fP an upper bound on the
increment in \fIposition\fP that would occur in a call to
MPI_Pack_external with the same values for \fIdatarep\fP,
\fIincount\fP, and \fIdatatype\fP.
.sp
The call returns an upper bound, rather than an exact bound, as the
exact amount of space needed to pack the message may depend on context
and alignment (e.g., the first message packed in a packing unit may
take more space).

.SH NOTES
.ft R
The \fIdatarep\fP argument specifies the data format. The only valid
value in the current version of MPI is "external32". The argument is
provided for future extensibility.

.SH ERRORS
.ft R
Almost all MPI routines return an error value; C routines as
the value of the function and Fortran routines in the last argument.
.sp
Before the error value is returned, the current MPI error handler is
called. By default, this error handler aborts the MPI job, except for
I/O function errors. The error handler may be changed with
MPI_Comm_set_errhandler; the predefined error handler MPI_ERRORS_RETURN
may be used to cause error values to be returned. Note that MPI does not
guarantee that an MPI program can continue past an error.
.sp
See the MPI man page for a full list of MPI error codes.

.SH SEE ALSO
.ft R
.nf
MPI_Pack_external
MPI_Unpack_external



doc/v1.4/man3/MPI_Pack_external_size.3.php
<?php
$topdir = "../../..";
$title = "MPI_Pack_external_size(3) man page (version 1.4.5)";
$meta_desc = "Open MPI v1.4.5 man page: MPI_PACK_EXTERNAL_SIZE(3)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
  <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<p>
<h2><a name='sect0' href='#toc0'>Name</a></h2>
<b>MPI_Pack_external_size</b> - Calculates upper bound on space needed to

<p>write to a portable format
<p>
<h2><a name='sect1' href='#toc1'>Syntax</a></h2>

<p>
<h2><a name='sect2' href='#toc2'>C Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
int MPI_Pack_external_size(char *datarep, int incount,
<tt> </tt>&nbsp;<tt> </tt>&nbsp;MPI_Datatype datatype, MPI_Aint *size)
</pre>
<h2><a name='sect3' href='#toc3'>Fortran Syntax</a></h2>
<br>
<pre>INCLUDE &rsquo;mpif.h&rsquo;
MPI_PACK_EXTERNAL_SIZE(DATAREP, INCOUNT, DATATYPE, SIZE, IERROR)
<tt> </tt>&nbsp;<tt> </tt>&nbsp;INTEGER<tt> </tt>&nbsp;<tt> </tt>&nbsp;<tt> </tt>&nbsp;<tt> </tt>&nbsp;INCOUNT, DATATYPE, IERROR
<tt> </tt>&nbsp;<tt> </tt>&nbsp;INTEGER (KIND=MPI_ADDRESS_KIND) SIZE
<tt> </tt>&nbsp;<tt> </tt>&nbsp;CHARACTER*(*)<tt> </tt>&nbsp;<tt> </tt>&nbsp;DATAREP
</pre>
<h2><a name='sect4' href='#toc4'>C++ Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
MPI::Aint MPI::Datatype::Pack_external_size(const char* datarep,
<tt> </tt>&nbsp;<tt> </tt>&nbsp;int incount) const
</pre>
<h2><a name='sect5' href='#toc5'>Input Parameters</a></h2>

<dl>

<dt>datarep </dt>
<dd>Data representation (string). </dd>

<dt>incount </dt>
<dd>Number of
input data items (integer). </dd>

<dt>datatype </dt>
<dd>Datatype of each input data item (handle).

<p> </dd>
</dl>

<h2><a name='sect6' href='#toc6'>Output Parameters</a></h2>

<dl>

<dt>size </dt>
<dd>Upper bound on size of packed message, in bytes
(integer). </dd>

<dt>IERROR </dt>
<dd>Fortran only: Error status (integer).
<p> </dd>
</dl>

<h2><a name='sect7' href='#toc7'>Description</a></h2>
MPI_Pack_external_size
allows the application to find out how much space is needed to pack a message
in the portable format defined by the MPI Forum. It returns in <i>size</i> an upper
bound on the increment in <i>position</i> that would occur in a call to <a href="../man3/MPI_Pack_external.3.php">MPI_Pack_external</a>
with the same values for <i>datarep</i>, <i>incount</i>, and <i>datatype</i>. <p>
The call returns
an upper bound, rather than an exact bound, as the exact amount of space
needed to pack the message may depend on context and alignment (e.g., the
first message packed in a packing unit may take more space).
<p>
<h2><a name='sect8' href='#toc8'>Notes</a></h2>
The <i>datarep</i>
argument specifies the data format. The only valid value in the current
version of MPI is "external32". The argument is provided for future extensibility.

<p>
<h2><a name='sect9' href='#toc9'>Errors</a></h2>
Almost all MPI routines return an error value; C routines as the
value of the function and Fortran routines in the last argument. C++ functions
do not return errors. If the default error handler is set to MPI::ERRORS_THROW_EXCEPTIONS,
then on error the C++ exception mechanism will be used to throw an MPI:Exception
object. <p>
Before the error value is returned, the current MPI error handler
is called. By default, this error handler aborts the MPI job, except for
I/O function errors. The error handler may be changed with <a href="../man3/MPI_Comm_set_errhandler.3.php">MPI_Comm_set_errhandler</a>;
the predefined error handler MPI_ERRORS_RETURN may be used to cause error
values to be returned. Note that MPI does not guarantee that an MPI program
can continue past an error.  <p>
See the MPI man page for a full list of MPI
error codes.
<p>
<h2><a name='sect10' href='#toc10'>See Also</a></h2>
<br>
<pre><a href="../man3/MPI_Pack_external.3.php">MPI_Pack_external</a>
<a href="../man3/MPI_Unpack_external.3.php">MPI_Unpack_external</a>

<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
